control-plane/agents/core/src/volume/specs.rs

```diff
@@ -570,6 +579,17 @@ impl ResourceSpecsLocked {
         let nexus =
             SpecOperations::validate_update_step(registry, result, &spec, &spec_clone).await?;
 
+        let (volume_id, last_nexus_id) = {
+            let volume_spec = spec.lock();
+            (volume_spec.uuid.clone(), volume_spec.last_nexus_id.clone())
+        };
+
+        // If there was a previous nexus we should delete the persisted NexusInfo structure.
+        if let Some(nexus_id) = last_nexus_id {
```

Comment:
we probably don't want to do this until after the `complete_update`?

Reply:
Good point. Done.